### PR TITLE
chipingress: add batching client with per-event partial delivery

### PR DESCRIPTION
**Pull request by pkcll**
## Summary

Adds proto types and client wiring for chipingress batch publishing with per-event partial-delivery semantics.

All wire-format changes are **additive and backwards-compatible** with `origin/main`.

## Proto (`pkg/chipingress/pb/chip_ingress.proto`)

- **`PublishOptions { optional bool transaction_enabled = 1 }`**
  - Unset or `false` ⇒ partial delivery (server default): valid events are produced, per-event errors are reported in `PublishResponse.results`.
  - Explicit `true` ⇒ all-or-nothing: any per-event failure fails the entire batch.
- **`CloudEventBatch.options = 2`** (`PublishOptions`).
- **`PublishResult.error = 2`** (`PublishError`).
- **`PublishError { PublishErrorCode error_code, string reason }`**.
- **`PublishErrorCode` enum**:
  - `PUBLISH_ERROR_CODE_UNKNOWN = 0`
  - `PUBLISH_ERROR_CODE_VALIDATION_FAILED = 1`
  - `PUBLISH_ERROR_CODE_SCHEMA_MISSING = 2`
  - `PUBLISH_ERROR_CODE_ENCODE_ERROR = 3`
  - `PUBLISH_ERROR_CODE_DOMAIN_MISCONFIGURATION = 4`
- **`PublishResult.eventId` → `event_id`** (same tag 1, same wire type, generated Go field `EventId` unchanged).

## Single-shot client (`pkg/chipingress/client.go`)

- `BatchOpt` + `WithTransactionEnabled(bool)` helper.
- `EventsToBatchWithOpts(events, opts...)` variadic constructor.
- `EventsToBatch` and `EventsToBatchWithOpts` default-populate `Options{TransactionEnabled: false}` so client intent is always explicit on the wire (defensive against any future server-default drift; server treats unset and explicit-false identically).
- Type aliases in `types.go`: `PublishOptions`, `PublishError`, `PublishErrorCode`.

## Batch client (`pkg/chipingress/batch`) — extend existing package

- Extends the existing batching client (`client.go` only; `batcher.go` and `stop_chan.go` untouched).
- New `transactionEnabled` field (default `false` = partial delivery); `Opt: WithTransactionEnabled(bool)`.
- `newBatchRequest` always emits `PublishOptions` explicitly (both `true` and `false`) matching the single-shot client contract.
- `splitMessagesByRequestSize` now sizes requests with `Options` populated so the byte budget matches the wire form.
- **Per-event callback dispatch** via new `completeBatchCallbacksFromResults`:
  - Partial-delivery path returns `*PublishError` per failed event.
  - Mismatch detection emits a synthetic `ErrCodeResultsMismatch` when `len(results) != len(messages)`.
  - Positional `event_id` mismatch is logged and counted via the `resultsMismatchTotal` metric.
- New `PublishError{Code chipingress.PublishErrorCode, Reason string}` with re-exported `ErrCode*` constants.

## Observability (OTel metrics)

- `configInfo` gauge gains a `transaction_enabled` bool attribute (reflects the effective client flag, `false` by default).
- New `resultsMismatchTotal` counter (incremented on positional `event_id` mismatch).
- Existing `sendRequestsTotal`, `requestSizeMessages`, `requestSizeBytes`, `requestLatencyMS`, `batchSplitsTotal` are unchanged.

## Tests

- Proto roundtrip: `PublishResult` with `PublishError`; `PublishOptions.GetTransactionEnabled` edge cases.
- Single-shot helper: default, `WithTransactionEnabled(true)`, `WithTransactionEnabled(false)`.
- Batch client: partial-delivery callbacks, RPC error fallback, `WithTransactionEnabled(true)` wire matcher, results-mismatch synthetic error, split-on-size with explicit-`Options` sizing.

## Compatibility vs `origin/main`

- **Public Go API**: strictly additive (no removed or renamed symbols).
- **Proto wire**: additive (new fields, same tag numbers; `eventId → event_id` is a name-only change, generated Go field `EventId` and binary wire unchanged).
- **Behavior**: `EventsToBatch` now returns a batch with `Options{TransactionEnabled: false}` populated where it was previously `nil`. Server treats both identically.

### CODE REVIEW

File: pkg/chipingress/pb/chip_ingress.proto [hunk 1]
```diff
@@ -28,17 +29,61 @@ service ChipIngress {
...
 
 message PublishResponse {
   repeated PublishResult results = 1;
 }
 
+// PublishResult reports the outcome for one input event.
+// Invariant: error is set if and only if the event was not produced to Kafka.
 message PublishResult {
-  string eventId = 1;
+  string event_id = 1;
```
Comment (pkcll) [inline pkg/chipingress/pb/chip_ingress.proto]: This is not a breaking change.

- Binary format: Protobuf serializes by field number, not name. Since the field number remains 1, existing binary-encoded messages are fully compatible.
- JSON format: Protobuf's JSON mapping uses the lowerCamelCase of the field name by default. event_id automatically maps to "eventId" in JSON — the same as the old eventId field name did.
 
- generated-code
  - Go: Both eventId and event_id produce the same generated accessor (EventId), so no breakage.

**Review by jmank88 — APPROVED** [review]

**Review by hendoxc — APPROVED** [review]
